File: src/mod2_collections/lesson24_arrays/ArrayDemo.java
package mod2_collections.lesson24_arrays;

import java.util.Arrays;

public class ArrayDemo {
    public static void main(String[] args) {

        //6 - сколько хранится элементов в массиве
        //в массиве нельзя изменить кол-во элементов
        int[] arraysNumber = new int[6];

        //ещё один вид инициализации
        int[] arraysNumber2 = {1, 2, 3, 4, 5, 6, 7, 2, 3, 45, 1};
        arraysNumber[0] = 4;
        arraysNumber[1] = 3;
        arraysNumber[2] = 5;
        arraysNumber[3] = 11;
        arraysNumber[4] = 2;
        arraysNumber[5] = 7;

        System.out.println(arraysNumber);
        System.out.println(Arrays.toString(arraysNumber));

        for (int i = 0; i < arraysNumber.length; i++) {
            System.out.println("The index is: " + i + ", the value is: " + arraysNumber[i]);
        }

    }
}

File: src/mod2_collections/lesson24_arrays/ArrayDemo.kt
package mod2_collections.lesson24_arrays

fun main() {

    //int[] arraysNumber2 = {1, 2, 3, 4, 5, 6, 7, 2, 3, 45, 1};
    val arrayNumbers = arrayOf(1, 2, 35, 46, 456, 234)

    //эквивалентно int[] arraysNumber = new int[6];
    val anotherWayArray = IntArray(2)

    println("Размер нашего массива = ${arrayNumbers.size}")

//    for (num in 0..23)
    for (num in arrayNumbers) {
        println("Значение: $num")
    }

    for ((index, value) in arrayNumbers.withIndex()){
        println("Индекс: $index; Значение: $value")
    }
}

